python/cudf/cudf/core/dataframe.py

```diff
@@ -3040,6 +3040,78 @@ def rename(
         else:
             return out.copy(deep=copy)
 
+    def add_prefix(self, prefix):
+        """
+        Prefix labels with string `prefix`.
+        For DataFrame, the column labels are prefixed.
+        Parameters
+        ----------
+        prefix : str
+            The string to add before each label.
+        Returns
+        -------
+        DataFrame
+            New DataFrame with updated labels.
+        See Also
+        --------
+        DataFrame.add_suffix: Suffix column labels with string `suffix`.
+        Examples
+        --------
+        >>> df = pd.DataFrame({'A': [1, 2, 3, 4], 'B': [3, 4, 5, 6]})
+        >>> df
+           A  B
+        0  1  3
+        1  2  4
+        2  3  5
+        3  4  6
+        >>> df.add_prefix('col_')
+             col_A  col_B
+        0       1       3
+        1       2       4
+        2       3       5
+        3       4       6
+        """
+        out = DataFrame(index=self.index)
+        out._data = self._data.copy(deep=False)
```

Comment:
Couple questions:

- Why the shallow data copy here?
- Is there any reason we opt for these two lines instead of:

```suggestion
        out = self.copy(deep=False)
```

Reply:
Thanks @charlesbluca for the reviews! Those are great questions, I was following the implementation of `rename` in the same file, and was confused about this as well. I assumed this to be a design choice. Let me know if you have better context on this :)

Reply:
Upon reviewing the code further, I realized it should indeed be a `deep` copy as we are returning a brand new `Series`/`DataFrame`. And this suggestion will work as well, and is probably better to do it in one line. Thanks!